### PR TITLE
Add a number of retries to the "reset guids" test

### PR DESCRIPTION
**Pull request by abdallahyas**
In an effort to reproduce the bug with the ib-sriov-cni where it
fails to reset VFs GUIDs after deletion. A number of retries to the
"reset guids" test in ib_sriov_cni_test.sh was added so that it
will increase the chance of the bug appearing.

### CODE REVIEW

File: sriov_ib_cni_test.sh [hunk 1]
```diff
@@ -142,35 +149,72 @@ function exit_code {
...
+
+status=0
 
-test_guids_reseted
+
+starting_guids=$(get_vfs_guids)
+
+echo ""
+echo "VFs GUIDs before the test are:"
+echo "$starting_guids"
```
Comment (adrianchiris) [inline sriov_ib_cni_test.sh]: Is there a reason you are printing `starting_guid` twice ?

File: sriov_ib_cni_test.sh [hunk 2]
```diff
@@ -142,35 +149,72 @@ function exit_code {
...
+     fi
 
-let status=status+$?
+     echo "all tests succeeded!!"
+
+     return $status
+    
+}
+
+status=0
```
Comment (adrianchiris) [inline sriov_ib_cni_test.sh]: if you are already refactoring here, can you put the test flow logic under "main" and just call main

File: sriov_ib_cni_test.sh [hunk 3]
```diff
@@ -142,35 +149,72 @@ function exit_code {
     exit $status
 }
 
-status=0
-echo "Creating pod test-pod-1"
-pod_create 'test-pod-1'
-let status=status+$?
+function test_pods {
```
Comment (adrianchiris) [inline sriov_ib_cni_test.sh:161]: This should really be a separate commit, but NVM

File: sriov_ib_cni_test.sh [hunk 4]
```diff
@@ -133,6 +136,10 @@ function test_guids_reseted {
     done
 }
 
+function get_vfs_guids {
```
Comment (adrianchiris) [inline sriov_ib_cni_test.sh:147]: you are only doing ip link show on `ib0`, should this be a parameter for the function ?

File: sriov_ib_cni_test.sh [hunk 5]
```diff
@@ -142,35 +149,72 @@ function exit_code {
...
-let status=status+$?
+    echo ""
+    echo "Pods created successfully! VFs GUIDs after pods creation are:"
+    after_creation_guids=$(get_vfs_guids)
+    echo "$after_creation_guids"
+    
+    test_pods_connectivity 'test-pod-1' 'test-pod-2'
+     
+    let status=status+$?
+
```
Comment (adrianchiris) [inline sriov_ib_cni_test.sh]: can you move the pod deleting logic from test_guids reseted ?

basically have:
1. delete pods
2. print guids after deletion
3. test guid reset

**Review by adrianchiris — CHANGES_REQUESTED** [review]

File: sriov_ib_cni_test.sh (hunk 4 above)
Comment (abdallahyas) [reply sriov_ib_cni_test.sh:147]: totally, will do it so that it gets this automatically, (the logic is already implemented in the install script, will do it here also)

File: sriov_ib_cni_test.sh (hunk 5 above)
Comment (abdallahyas) [reply sriov_ib_cni_test.sh]: will do

File: sriov_ib_cni_test.sh (hunk 2 above)
Comment (abdallahyas) [reply sriov_ib_cni_test.sh]: will do

File: sriov_ib_cni_test.sh (hunk 1 above)
Comment (abdallahyas) [reply sriov_ib_cni_test.sh]: i print all the guids two times:
    - When the guid is changed, to give more information at the same location the guid changed.
    - At the end of the test, guids for all stages are printed as a summary to ease the debugging.

Comment (abdallahyas) [reply sriov_ib_cni_test.sh]: basically, i print the guids twice:
* When it first change, to have more information at the place it happened.
* At the end of the test to have a summary of all the guids during the test (easier to debug).

File: sriov_ib_cni_test.sh (hunk 4 above)
Comment (abdallahyas) [reply sriov_ib_cni_test.sh:147]: This is done, but i think because i did not change this same line, it is not marked as outdated.

**Comment by abdallahyas** [comment]
@adrianchiris  This is ready.

File: sriov_ib_cni_test.sh [hunk 6]
```diff
@@ -142,35 +158,86 @@ function exit_code {
...
+    
+}
 
-if [ "$status" != 0 ]; then
-    echo "Error: error in testing the pods"
+function main {
+    status=0
+
+    if [ $SRIOV_INTERFACE == 'auto_detect' ]; then
+        export SRIOV_INTERFACE=$(ls -l /sys/class/net/ | grep $(lspci |grep Mellanox | grep MT27800|head -n1|awk '{print $1}') | awk '{print $9}')
```
Comment (adrianchiris) [inline sriov_ib_cni_test.sh:218]: cant this be a parameter provided to the script by jenkins ?

Comment (adrianchiris) [inline sriov_ib_cni_test.sh]: lets exclude cx3/cx3pro instead of working with cx4 only on autodetect

**Review by adrianchiris — APPROVED** [review]
